pkg/autoscaler/autoscaler.go

```diff
@@ -90,7 +90,7 @@ func (a *Autoscaler) Reconcile() error {
 	if err != nil {
 		return err
 	}
-	a.Status.CurrentReplicas = scaleTarget.Status.Replicas
+	a.Status.CurrentReplicas = &scaleTarget.Status.Replicas
```

Comment:
Can `scaleTarget.Status.Replicas` ever be NIL? If it ever happens it will panic.

Reply:
scaleTarget.Status.Replicas is an int32
```
type ScaleStatus struct {
	// actual number of observed instances of the scaled object.
	Replicas int32 `json:"replicas" protobuf:"varint,1,opt,name=replicas"`

	// label query over pods that should match the replicas count. This is same
	// as the label selector but in the string format to avoid introspection
	// by clients. The string will be in the same format as the query-param syntax.
	// More info about label selectors: http://kubernetes.io/docs/user-guide/labels#label-selectors
	// +optional
	Selector string `json:"selector,omitempty" protobuf:"bytes,2,opt,name=selector"`
}
```

Reply:
My bad, was looking at the wrong thing.